### PR TITLE
[MOBILE-696] Update to latest UA Android SDK 10.0.2

### PR DESCRIPTION
**Pull request by BrianBatchelder**
Updated SDK dependency to 10.0.2
Fixed build errors due to API changes.

It might be more interesting to review it commit by commit. The bulk of the changes were the parts of the sample app that were auto-created by Android Studio.

### CODE REVIEW

File: segment-integration/src/main/java/com/urbanairship/extension/segment/SegmentAutopilot.java [hunk 1]
```diff
@@ -50,11 +49,6 @@ public AirshipConfigOptions createAirshipConfigOptions(@NonNull Context context)
                     .setProductionAppSecret(preferences.getString(APP_SECRET, null));
         }
 
-        // Apply GCM Sender from Segment
```
Comment (rlepinski) [inline segment-integration/src/main/java/com/urbanairship/extension/segment/SegmentAutopilot.java]: You still want to respect this. `builder.setFcmSender`

Comment (BrianBatchelder) [reply segment-integration/src/main/java/com/urbanairship/extension/segment/SegmentAutopilot.java]: Done.

File: sample/build.gradle [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    // Segment
+    implementation 'com.segment.analytics.android:analytics:4.+'
+
+    // Airship Integration
+    // apps add integration using the following dependency 
+//    implementation 'com.urbanairship.android:segment-integration:+'
+    //  only this sample app uses the following local dependency
+    implementation project(":segment-integration")  
+
+    implementation 'com.urbanairship.android:urbanairship-core:10.0.2'
```
Comment (BrianBatchelder) [inline sample/build.gradle]: I don't understand why I had to add this dependency to access the core. Shouldn't I get this through the segment-integration dependency? I tried `api` also, but no luck.

Comment (marc-scig) [reply sample/build.gradle]: Did you try using api on the segment-integration build.gradle or just here? I think that's the place it would matter.

Comment (BrianBatchelder) [reply sample/build.gradle]: Cleaned up.

File: README.md [hunk 3]
```diff
@@ -1,43 +1,72 @@
...
+```
+
+5) Set up a Push Provider as described in the following [Airship docs](https://docs.airship.com/platform/android/getting-started/#push-provider-setup). Add the Airship core and your chosen push provider as a dependency in your app's build.gradle file. e.g. for FCM:
+```
+   dependencies {
        ...
-       .build();
+
+       // Airship Core
+       implementation 'com.urbanairship.android:urbanairship-core:10.0.2'
```
Comment (marc-scig) [inline README.md]: They shouldn't need to depend on core if they're pulling in fcm, right?

Comment (BrianBatchelder) [reply README.md]: Nope!

**Pull request merged**